backend/clickhouse/trace_row.go

```diff
@@ -0,0 +1,121 @@
+package clickhouse
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/highlight-run/highlight/backend/model"
+	e "github.com/pkg/errors"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type TraceRow struct {
+	ProjectId          uint32
+	SecureSessionId    string
+	Timestamp          time.Time
+	TraceId            string
+	SpanId             string
+	ParentSpanId       string
+	TraceState         string
+	SpanName           string
+	SpanKind           string
+	ServiceName        string
+	ResourceAttributes map[string]string
+	SpanAttributes     map[string]string
+	Duration           int64
+	StatusCode         string
+	StatusMessage      string
+	Events             []Event
+	Links              []Link
+}
+
+type Event struct {
+	Timestamp  time.Time         `json:"timestamp"`
+	Name       string            `json:"name"`
+	Attributes map[string]string `json:"attributes,omitempty"`
+}
+
+type Link struct {
+	TraceId    string            `json:"trace_id"`
+	SpanId     string            `json:"span_id"`
+	TraceState string            `json:"trace_state"`
+	Attributes map[string]string `json:"attributes,omitempty"`
+}
+
+func NewTraceRow(span ptrace.Span) *TraceRow {
+	attributes := attributesToMap(span.Attributes().AsRaw())
+
+	// TODO: Use extractFields instead of copying logic here. For some reason we
+	// can't import from extract.go.
+	projectId, err := projectToInt(attributes["highlight.project_id"])
```

Comment:
@et n00b Go question here, but do you know why I couldn't import `extractFields` to this file? I wanted to use that to get the project and session IDs here.

Reply:
You have a circular dependency. 

You shouldn't need to extract the projectID, it's done for you in `otel.go`.

https://github.com/highlight/highlight/blob/0a70f450b0c94ce0da69b8d56e4786c1704ef4d6/backend/otel/otel.go#L177-L178

I don't think `NewTraceRow` should have any knowledge of otel related information (e.g.  `ptrace.Span`) as that should be handled up the chain.

Reply:
Thanks, @et! I've refactored this so we don't reference any otel logic here and made helpers as function methods, similar to what we have in `log_row.go`. Appreciate your feedback on this!